src/plugins/index_pattern_management/public/components/index_pattern_table/empty_state/empty_state.tsx

```diff
@@ -0,0 +1,234 @@
+import {
+  EuiPageContentHeader,
+  EuiPageContentHeaderSection,
+  EuiTitle,
+  EuiPageContentBody,
+  EuiPageContent,
+  EuiIcon,
+  EuiSpacer,
+  EuiFlexItem,
+  EuiDescriptionList,
+  EuiFlexGrid,
+  EuiCard,
+  EuiLink,
+  EuiText,
+} from '@elastic/eui';
+import { useHistory } from 'react-router-dom';
+import { reactRouterNavigate } from '../../../../../../plugins/kibana_react/public';
+import { MlCardState } from '../../../types';
+
+export const EmptyState = ({
+  onRefresh,
+  navigateToApp,
+  docLinks,
+  getMlCardState,
+  canSave,
+}: {
+  onRefresh: () => void;
+  navigateToApp: ApplicationStart['navigateToApp'];
+  docLinks: DocLinksStart;
+  getMlCardState: () => MlCardState;
+  canSave: boolean;
+}) => {
+  const mlCard = (
+    <EuiFlexItem>
+      <EuiCard
+        onClick={() => navigateToApp('ml', { path: '#/filedatavisualizer' })}
+        className="inpEmptyState__card"
+        betaBadgeLabel={
+          getMlCardState() === MlCardState.ENABLED
+            ? undefined
+            : i18n.translate(
+                'indexPatternManagement.createIndexPattern.emptyState.basicLicenseLabel',
+                {
+                  defaultMessage: 'Basic',
+                }
+              )
+        }
+        betaBadgeTooltipContent={i18n.translate(
+          'indexPatternManagement.createIndexPattern.emptyState.basicLicenseDescription',
+          {
+            defaultMessage: 'This feature requires a Basic license.',
+          }
+        )}
+        isDisabled={getMlCardState() === MlCardState.DISABLED}
+        icon={<EuiIcon size="xl" type="document" color="subdued" />}
+        title={
+          <FormattedMessage
+            id="indexPatternManagement.createIndexPattern.emptyState.uploadCardTitle"
+            defaultMessage="Upload a file"
+          />
+        }
+        description={
+          <FormattedMessage
+            id="indexPatternManagement.createIndexPattern.emptyState.uploadCardDescription"
+            defaultMessage="Import a CSV, NDJSON, or log file."
+          />
+        }
+      />
+    </EuiFlexItem>
+  );
+
+  const createAnyway = (
+    <EuiText color="subdued" textAlign="center" size="xs">
+      <FormattedMessage
+        id="indexPatternManagement.createIndexPattern.emptyState.createAnyway"
+        defaultMessage="Some indices may be hidden. Try to {link} anyway."
+        values={{
+          link: (
+            <EuiLink {...reactRouterNavigate(useHistory(), 'create')} data-test-subj="createAnyway">
+              <FormattedMessage
+                id="indexPatternManagement.createIndexPattern.emptyState.createAnywayLink"
+                defaultMessage="create an index pattern"
+              />
+            </EuiLink>
+          ),
+        }}
+      />
+    </EuiText>
+  );
+
+  return (
+    <>
+      <EuiPageContent
+        className="inpEmptyState"
+        grow={false}
+        horizontalPosition="center"
+        data-test-subj="emptyState"
```

Comment:
nit: `emptyState` is a pretty vague test subject ID which I could see someone else accidentally using in the future; you might consider making this more specific like `indexPatternEmptyState`